KArray/KArrayCompile.lean

```diff
@@ -5,17 +5,40 @@ open Lean
 initialize kArrayCompileAttr : TagAttribute ←
   registerTagAttribute `kcompile "tag to request KArray compile"
 
-constant cHeader : String :=
-  "#include <lean/lean.h>\n"
+constant cIncludes : String :=
+  "#include <lean/lean.h>"
 
-/- The magic happens here -/
-def mkCCode (targetName : Name) (expr : Expr) : String :=
-  s!"{targetName}|{expr}"
+constant cDefines : String :=
+  "#define external extern \"C\" LEAN_EXPORT"
 
--- TODO: make sure `targetName` is a valid function name for C code
--- TODO: make sure no `targetName` is ever duplicated
-def extractCCodeFromEnv (env : Environment) : IO String := do
-  let mut res ← ""
+constant semicolon : String := ";"
+
+constant newLine : String := "\n"
+
+constant semicolonNewLine : String := semicolon ++ newLine
```

Comment:
Why are there `constant` I understand that these are 'constants' in the meaning of the word but why are they Lean's constants? `constant` in Lean is an opaque definition, you can argue about it you just postulate that it exists. It does not really matter, because we are not going to prove anything about this part of the code but I find it a bit confusing.